zeppelin-jupyter/src/main/java/org/apache/zeppelin/jupyter/JupyterUtil.java

```diff
@@ -0,0 +1,203 @@
+
+  private final RuntimeTypeAdapterFactory<Cell> cellTypeFactory;
+  private final RuntimeTypeAdapterFactory<Output> outputTypeFactory;
+
+  public JupyterUtil() {
+    this.cellTypeFactory = RuntimeTypeAdapterFactory.of(Cell.class, "cell_type")
+        .registerSubtype(MarkdownCell.class, "markdown").registerSubtype(CodeCell.class, "code")
+        .registerSubtype(RawCell.class, "raw");
+    this.outputTypeFactory = RuntimeTypeAdapterFactory.of(Output.class, "output_type")
+        .registerSubtype(ExecuteResult.class, "execute_result")
+        .registerSubtype(DisplayData.class, "display_data")
+        .registerSubtype(Stream.class, "stream").registerSubtype(Error.class, "error");
+  }
+
+  public Nbformat getNbformat(Reader in) {
+    return getNbformat(in, new GsonBuilder());
+  }
+
+  public Nbformat getNbformat(Reader in, GsonBuilder gsonBuilder) {
+    return getGson(gsonBuilder).fromJson(in, Nbformat.class);
+  }
+
+  public Note getNote(Reader in, String codeReplaced, String markdownReplaced) {
+    return getNote(in, new GsonBuilder(), codeReplaced, markdownReplaced);
+  }
+
+  public Note getNote(Reader in, GsonBuilder gsonBuilder, String codeReplaced,
+      String markdownReplaced) {
+    return getNote(getNbformat(in, gsonBuilder), codeReplaced, markdownReplaced);
+  }
+
+  public Note getNote(Nbformat nbformat, String codeReplaced, String markdownReplaced) {
+    Note note = new Note();
+
+    String name = nbformat.getMetadata().getTitle();
+    if (null == name) {
+      name = "Note converted from Jupyter";
+    }
+    note.setName(name);
+
+    String lineSeparator = System.lineSeparator();
+    Paragraph paragraph;
+    List<Paragraph> paragraphs = new ArrayList<>();
+    String interpreterName;
+    List<TypeData> typeDataList;
+    String type;
+    String result;
+
+    for (Cell cell : nbformat.getCells()) {
+      paragraph = new Paragraph();
+      typeDataList = new ArrayList<>();
+
+      if (cell instanceof CodeCell) {
+        interpreterName = "%" + codeReplaced;
+        for (Output output : ((CodeCell) cell).getOutputs()) {
+          TypeData typeData;
+          if (output instanceof Stream) {
+            type = TypeData.TEXT;
+            result = Joiner.on(lineSeparator).join(((Stream) output).getText());
+            typeData = new TypeData(type, result);
+            typeDataList.add(typeData);
+          } else if (output instanceof ExecuteResult || output instanceof DisplayData) {
+            Map<String, Object> data =
+                (output instanceof ExecuteResult) ? ((ExecuteResult) output).getData()
+                    : ((DisplayData) output).getData();
+            for (Map.Entry<String, Object> datum : data.entrySet()) {
+              if (TEXT_PLAIN.equals(datum.getKey())) {
+                type = TypeData.TEXT;
+                result = Joiner.on(lineSeparator).join((List<String>) datum.getValue());
+              } else if (IMAGE_PNG.equals(datum.getKey())) {
```

Comment:
Thanks. I've found these kinds of docs. I referred the office nb format docs which doesn't limit types. I have a question. In case of Jupyter, it always seems to stored images as png type. doesn't it?